src/request/mod.rs

```diff
@@ -79,13 +80,21 @@ impl<'a, 'b> Request<'a, 'b> {
             },
 
             AbsolutePath(ref path) => {
-                // Attempt to prepend the Host header (mandatory in HTTP/1.1)
-                let url_string = match headers.get::<headers::Host>() {
-                    Some(ref host) => {
-                        format!("{}://{}:{}{}", protocol.name(), host.hostname, local_addr.port(),
-                                path)
+                let url_string = match (version, headers.get::<headers::Host>()) {
+                    (_, Some(ref host)) => {
+                        // Attempt to prepend the Host header (mandatory in HTTP/1.1)
+                        format!("{}://{}:{}{}", protocol.name(), host.hostname, local_addr.port(), path)
                     },
-                    None => return Err("No host specified in request".to_string())
+                    (v, None) if v < HttpVersion::Http11 => {
+                        // Attempt to use the local address? (host header is not required in HTTP/1.0).
+                        match local_addr {
+                            SocketAddr::V4(addr4) => format!("{}://{}:{}{}", protocol.name(), addr4.ip(), local_addr.port(), path),
```

Comment:
I'm kind of unhappy to write addr serialization code here. Isn't there something in `url` that we can use for this?